docs/design/2022-01-04-integer-shard-index.md

```diff
@@ -0,0 +1,423 @@
+# Integer shard index
+
+- RFC PR: https://github.com/tikv/rfcs/pull/0084
+- Tracking Issue: https://github.com/pingcap/tidb/issues/31040
+
+## Summary
```

Comment:
More descriptions or examples of the typical user case could be added here, such as:
```
create table t(id int primary key auto_increment)；
...
```